sfepy/discrete/common/dof_info.py

```diff
@@ -346,14 +361,21 @@ def map_equations(self, bcs, field, ts, functions, problem=None,
 
             active_bcs.add(bc.key)
 
-            if isinstance(bc, EssentialBC):
+            if isinstance(bc, DGEssentialBC):
```

Comment:
Just from the "new feature to the end" perspective, I would put the DG conditions after the FEM ones.

Reply:
There is an issue with class hierarchy `DGEssentialBC` is subclass of `EssentialBC`, hence it is also instance of it.

Reply:
I see. Well, one could use `type(bc) == ...` but the cleanest way would be to make `ntype` and `region` (for periodic BCs) attributes of `bc` in the constructors in `conditions.py`. Then it would be more pythonic and the isinstance tests would not be needed. But in this PR we could just leave it as it is.